File: app/src/main/java/com/govnokoder/velotracker/ui/main/ViewPagerAdapter.java
package com.govnokoder.velotracker.ui.main;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.govnokoder.velotracker.R;

public class ViewPagerAdapter extends FragmentStateAdapter {


    public ViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return(PageStart.newInstance(position));
            case 1:
                return(PageHistory.newInstance(position));
            case 2:
                return(PageStatistics.newInstance(position));
        }
        return(PageStart.newInstance(position));
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: app/src/main/java/com/coursework/velotracker/ui/main/ViewPagerAdapter.kt
package com.coursework.velotracker.ui.main

import androidx.annotation.NonNull
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class ViewPagerAdapter(fragmentActivity: FragmentActivity): FragmentStateAdapter(fragmentActivity) {


    override fun getItemCount(): Int {
        return 3
    }

    override fun createFragment(position: Int): Fragment {
        TODO()
//        switch (position) {
//            case 0:
//            return(PageStart.newInstance(position));
//            case 1:
//            return(PageHistory.newInstance(position));
//            case 2:
//            return(PageStatistics.newInstance(position));
//        }
//        return(PageStart.newInstance(position));
    }

    override fun getItemId(position: Int): Long {
        return 3
    }


}